models/nc-news.models.js

```diff
@@ -22,24 +22,38 @@ const fetchArticleById = (article_id) => {
     })
 }
 
-const fetchArticles = () => {
-    return db.query(
-        `SELECT
-        articles.author,
-        articles.title,
-        articles.article_id,
-        articles.topic,
-        articles.created_at,
-        articles.votes,
-        articles.article_img_url, 
-        COUNT(comments.article_id) 
-        AS comment_count 
-        FROM articles 
-        LEFT JOIN comments 
-        ON comments.article_id = articles.article_id 
-        GROUP BY articles.article_id 
-        ORDER BY created_at DESC`)
+const fetchArticles = (topic) => {
+    let queryStr = `
+    SELECT
+    articles.author,
+    articles.title,
+    articles.article_id,
+    articles.topic,
+    articles.created_at,
+    articles.votes,
+    articles.article_img_url, 
+    COUNT(comments.article_id) 
+    AS comment_count 
+    FROM articles 
+    LEFT JOIN comments 
+    ON comments.article_id = articles.article_id`;
+
+    const params = [];
+
+    if (topic) {
+        queryStr += ` WHERE articles.topic = $1`;
+        params.push(topic);
+    }
+
+    queryStr += ` 
+    GROUP BY articles.article_id 
+    ORDER BY created_at DESC;`;
+
```

Comment:
Great use of string concatenation to get the query to accept an optional topic 👍